db/pebble/db.go

```diff
@@ -109,3 +113,13 @@ func (d *DB) Update(fn func(txn db.Transaction) error) error {
 func (d *DB) Impl() any {
 	return d.pebble
 }
+
+func discardTxnOnPanic(txn db.Transaction) {
+	p := recover()
+	if p != nil {
+		if err := txn.Discard(); err != nil {
```

Comment:
What happens now when function triggers panic inside db layer? I guess it's going to be not commited anyway, right? Or it causes memory leak?

Reply:
`txn.Discard()` panicing is undefined behaviour. Node will try to shutdown gracefully but may corrupt the database.